internal/cephfs/store/fsjournal.go

```diff
@@ -128,7 +128,20 @@ func CheckVolExists(ctx context.Context,
 		}
 		err = cloneState.ToError()
 		if errors.Is(err, cerrors.ErrCloneInProgress) {
-			return nil, err
+			progressReport := cloneState.GetProgressReport()
+			if progressReport.PercentageCloned == "" || progressReport.AmountCloned == "" ||
+				progressReport.FilesCloned == "" {
+				return nil, err
+			} else {
+				err = fmt.Errorf("%w. progress report: percentage cloned=%s, amount cloned=%s, files cloned=%s",
+					err,
+					progressReport.PercentageCloned,
+					progressReport.AmountCloned,
+					progressReport.FilesCloned)
+				log.UsefulLog(ctx, err.Error())
+
+				return nil, err
+			}
```

Comment:
```suggestion
			progressReport := cloneState.GetProgressReport()
			// append progress report only if all the parameters are present.
			if (progressReport.PercentageCloned == "" || progressReport.AmountCloned == "" ||
				progressReport.FilesCloned == "") {
				err = fmt.Errorf("%w. progress report: percentage cloned=%s, amount cloned=%s, files cloned=%s",
					err,
					progressReport.PercentageCloned,
					progressReport.AmountCloned,
					progressReport.FilesCloned)
			}
			log.ErrorLog(ctx, err.Error())

			return nil, err
```

Reply:
@Nikhil-Ladha 

github should not allow me to comment if i have pending review comments  
:facepalm:
sorry for the confusion.

Reply:
I was wondering that if one is not empty so neither of them would be, so we can just add a check for `PercentageCloned` instead of all 3 params. WDYT @Madhu-1 @Rakshith-R ?

Reply:
This approach works for me.

Reply:
Updated the code accordingly.

Reply:
@Madhu-1 if we are good with this logic, can we go ahead and merge this?